core/libp2p/muxer/muxed_connection_config.hpp

```diff
@@ -13,11 +13,28 @@ namespace libp2p::muxer {
    * Config of muxed connection
    */
   struct MuxedConnectionConfig {
+    static constexpr uint32_t DEFAULT_MAXIMUM_WINDOW_SIZE = 1024u * 1024u;
+    static constexpr uint32_t DEFAULT_MAXIMUM_STREAMS = 1000u;
+
+    /**
+     * @brief construcsts configuration instance
+     * @param max_window_size how much unconsumed data each stream can have
+     * stored locally
+     * @param max_streams how much streams can be supported by Yamux at one time
+     */
+    inline MuxedConnectionConfig(uint32_t max_window_size, uint32_t max_streams)
+        : maximum_window_size{max_window_size}, maximum_streams{max_streams} {}
+
+    static MuxedConnectionConfig makeDefault() {
+      return MuxedConnectionConfig(DEFAULT_MAXIMUM_WINDOW_SIZE,
+                                   DEFAULT_MAXIMUM_STREAMS);
+    }
```

Comment:
hm. this does not sound right.
as I recall, the issue was that injector specifies default value (of 0) to constructor arguments.

Better way to solve this is to create a class `Named` which is NOT default constructible, and templated. Something like that:
```
template <typename T, typename Tag>
struct Named {
  using tag_type = Tag;
  using value_type = T;
  Named() = delete;
  Named(T t): value_(t) {}
  // also overload operator =, copy and move constructors

  T unwrap(){ return value_; }
private:
  T value_;
}
```

Then do 
```
struct MaxWindowSizeTag;
using MaxWindowSizeType = Named<uint32_t, MaxWindowSizeTag>;

struct MaxStreamsTag;
using MaxStreamsSizeType = Named<uint32_t, MaxStreamsTag>;
```

and 

```
struct MuxedConnectionConfig {
  MaxWindowSIzeType max_window_size = 1000;
  MaxStreamsSizeType max_streams = 1000;
}
```

Reply:
It doesn't work. boost::di is able to initialize even this. Probably we need to add a mechanism to ensure that configuration was specified or mention it in docs and keep in mind. It works right if I specify it like follows: 
```
auto inj = injector::makeNetworkInjector(
      boost::di::bind<muxer::MuxedConnectionConfig>.to(
          muxer::MuxedConnectionConfig()));
auto cfg = inj.create<muxer::MuxedConnectionConfig>();
```
If default constructor is defined of course. For now I will remove named and leave default and normal constructors.